loclist.kak

```diff
@@ -0,0 +1,147 @@
+## Location List
+
+declare-option -hidden str loclist_buffer
+declare-option -hidden str-list loclist_stack
+declare-option -docstring %{Defines location patterns for location lists.
+Default locations look like "file:line[:column][:message]"
+
+Capture groups must be:
+    1: filename
+    2: line number
+    3: optional column
+    4: optional message
+} regex loclist_location_format ^\h*\K([^:\n]+):(\d+)\b(?::(\d+)\b)?(?::([^\n]+))
+
+declare-option str loclist_root ''
```

Comment:
A related problem is that `*git*` buffers don't capture the Git worktree where they came from. There is a proposed Kakoune patch to fix that for `*git*`.
It's fine to duplicate this; this is mainly a note to self to make sure these two are consistent.

Maybe we should associate all scratch buffers with a directory but probably not.